src/runners/http_rfqm_service_runner.ts

```diff
@@ -58,18 +62,35 @@ if (require.main === module) {
             meshHttpUri: undefined,
         };
         const connection = await getDBConnectionAsync();
-        const rfqmService = await buildRfqmServiceAsync(connection);
+        const rfqmService = await buildRfqmServiceAsync(connection, false);
         const configManager = new ConfigManager();
-
         await runHttpRfqmServiceAsync(rfqmService, configManager, config, connection);
     })().catch((error) => logger.error(error.stack));
 }
 
 /**
  * Builds an instance of RfqmService
  */
-export async function buildRfqmServiceAsync(connection: Connection): Promise<RfqmService> {
-    const provider = providerUtils.createWeb3Provider(defaultHttpServiceWithRateLimiterConfig.ethereumRpcUrl);
+export async function buildRfqmServiceAsync(connection: Connection, asWorker: boolean): Promise<RfqmService> {
+    let provider: SupportedProvider;
+
+    if (asWorker) {
+        if (META_TX_WORKER_MNEMONIC === undefined) {
+            throw new Error(`META_TX_WORKER_MNEMONIC must be defined to run RFQM service as a worker`);
+        }
+        if (META_TX_WORKER_INDEX === undefined) {
+            throw new Error(`META_TX_WORKER_INDEX must be defined to run RFQM service as a worker`);
+        }
+        const workerPrivateKey = RfqBlockchainUtils.getPrivateKeyFromIndexAndPhrase(
+            META_TX_WORKER_MNEMONIC,
+            META_TX_WORKER_INDEX,
+        );
+        const privateWalletSubprovider = new PrivateKeyWalletSubprovider(workerPrivateKey);
+        const rpcProvider = providerUtils.createWeb3Provider(defaultHttpServiceWithRateLimiterConfig.ethereumRpcUrl);
+        provider = RfqBlockchainUtils.createPrivateKeyProvider(rpcProvider, privateWalletSubprovider);
+    } else {
+        provider = providerUtils.createWeb3Provider(defaultHttpServiceWithRateLimiterConfig.ethereumRpcUrl);
```

Comment:
Small suggestion: Why not factor this out of the if/else loop? it looks like it's duplicated